go/vt/vtctl/workflow.go

```diff
@@ -0,0 +1,174 @@
+package vtctl
+
+import (
+	"flag"
+	"fmt"
+
+	"golang.org/x/net/context"
+
+	"github.com/youtube/vitess/go/vt/servenv"
+	"github.com/youtube/vitess/go/vt/workflow"
+	"github.com/youtube/vitess/go/vt/wrangler"
+)
+
+// This file contains the workflows command group for vtctl.
+
+const workflowsGroupName = "Workflows"
+
+var (
+	// WorkflowManager contains our manager. It needs to be set before
+	// servenv.Run is called, or this command group will be disabled.
+	WorkflowManager *workflow.Manager
+)
+
+func init() {
+	servenv.OnRun(func() {
+		if WorkflowManager == nil {
+			return
+		}
+
+		addCommandGroup(workflowsGroupName)
+
+		addCommand(workflowsGroupName, command{
+			"WorkflowCreate",
+			commandWorkflowCreate,
+			"[-nostart] <factoryName> [parameters...]",
+			"Creates the workflow with the provided parameters. The workflow is also started, unless -nostart is specified."})
+		addCommand(workflowsGroupName, command{
+			"WorkflowStart",
+			commandWorkflowStart,
+			"<uuid>",
+			"Starts the workflow."})
+		addCommand(workflowsGroupName, command{
+			"WorkflowStop",
+			commandWorkflowStop,
+			"<uuid>",
+			"Stops the workflow."})
+		addCommand(workflowsGroupName, command{
+			"WorkflowWait",
+			commandWorkflowWait,
+			"<uuid>",
+			"Waits for the workflow to finish."})
+
+		addCommand(workflowsGroupName, command{
+			"WorkflowTree",
+			commandWorkflowTree,
+			"",
+			"Displays a JSON representation of the workflow tree."})
+		addCommand(workflowsGroupName, command{
+			"WorkflowAction",
+			commandWorkflowAction,
+			"<path> <name>",
+			"Sends the provided action name on the specified path."})
+	})
+}
+
+func commandWorkflowCreate(ctx context.Context, wr *wrangler.Wrangler, subFlags *flag.FlagSet, args []string) error {
+	if WorkflowManager == nil {
```

Comment:
This command is registered in servenv.OnRun(), so nobody should come here before that, and by then WorkflowManager is either initialized or will never be. I agree we have to be careful in tests.